pithos/pithos.py

```diff
@@ -366,6 +367,23 @@ def get_proxy(self):
 
         return None
 
+    def on_explicit_content_filter_checkbox(self, checkbox_state):
+        if self.connected:
+            if self.old_filter_state != checkbox_state:
+                self.pandora.set_explicit_content_filter_to(checkbox_state)
```

Comment:
This function does blocking IO, you would need to use `self.worker.run()` like elsewhere in the codebase.